pkg/combustion/network.go

```diff
@@ -0,0 +1,151 @@
+package combustion
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/template"
+	"go.uber.org/zap"
+)
+
+const (
+	nmcExecutable           = "nmc"
+	nmcConfigDir            = "config"
+	networkConfigDir        = "network"
+	networkConfigScriptName = "configure-network.sh"
+)
+
+//go:embed templates/configure-network.sh.tpl
+var configureNetworkScript string
+
+// Configures the network component if enabled.
+//
+//  1. Generates network configurations
+//  2. Copies the NMC executable
+//  3. Writes the configuration script
+//
+// Example result file layout:
+//
+//	combustion
+//	├── network
+//	│   ├── config
+//	│   │   ├── node1.example.com
+//	│   │   │   ├── eth0.nmconnection
+//	│   │   │   └── eth1.nmconnection
+//	│   │   ├── node2.example.com
+//	│   │   │   └── eth0.nmconnection
+//	│   │   ├── node3.example.com
+//	│   │   │   ├── bond0.nmconnection
+//	│   │   │   └── eth1.nmconnection
+//	│   │   └── host_config.yaml
+//	│   └── nmc
+//	└── configure-network.sh
+func configureNetwork(ctx *image.Context) ([]string, error) {
+	networkDir, err := generateComponentPath(ctx, networkConfigDir)
+	if err != nil {
+		return nil, fmt.Errorf("generating network path: %w", err)
+	}
+
+	if networkDir == "" {
+		// Networking is not configured, exit early
+		return nil, nil
+	}
+
+	zap.L().Info("Configuring network component...")
+
+	if err = generateNetworkConfig(ctx, networkDir); err != nil {
+		return nil, fmt.Errorf("generating network config: %w", err)
+	}
+
+	if err = writeNMCExecutable(ctx); err != nil {
+		return nil, fmt.Errorf("writing nmc executable: %w", err)
+	}
+
+	scriptName, err := writeNetworkConfigurationScript(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("writing network configuration script: %w", err)
+	}
+
+	zap.L().Info("Successfully configured network component")
+
+	return []string{scriptName}, nil
+}
+
+func generateNetworkConfig(ctx *image.Context, configDir string) error {
+	logFilename := generateNetworkLogFilename(ctx)
+	logFile, err := os.Create(logFilename)
+	if err != nil {
+		return fmt.Errorf("creating log file: %w", err)
+	}
+
+	defer func() {
+		if err = logFile.Close(); err != nil {
+			zap.L().Warn("Failed to close network log file properly", zap.Error(err))
+		}
+	}()
+
+	combustionNetworkDir := filepath.Join(ctx.CombustionDir, networkConfigDir, nmcConfigDir)
+
+	cmd := exec.Command(nmcExecutable, "generate",
+		"--config-dir", configDir,
+		"--output-dir", combustionNetworkDir)
+	cmd.Stdout = logFile
+	cmd.Stderr = logFile
+
+	if err = cmd.Run(); err != nil {
+		return fmt.Errorf("running nmc generate: %w", err)
+	}
+
+	return nil
+}
+
+func generateNetworkLogFilename(ctx *image.Context) string {
+	const networkConfigLogFile = "network-config-%s.log"
+
+	timestamp := time.Now().Format("Jan02_15-04-05")
```

Comment:
This uses the same format as the xorriso log file, which I like for consistency (I don't have particularly strong feelings right now about what the format looks like and didn't love it at the time). I know you mentioned an issue with constants, but part of me likes the idea of pulling this sort of thing out into a util--- UM, I mean, single timestamp format across EIB.

Reply:
We should probably think about how we want to organise the build directory itself.

Aside from the combustion subdirectory, it'd have a lot of files flying around and once more and more optional components start adding their own .log (and other) files, then the directory itself would become fairly cluttered. I've already seen this where I have around ~10 `iso-build` logs and around ~5 `network-config` logs. And all runs actually seem to override the combustion subdirectory as well.

Perhaps having a `build-<timestamp>` subdirectory which stores all particular files of a single build makes sense. This way we probably would not need timestamps for each log file at all.

Once we figure this out we can look into what package structure to use to unify all of the relevant functionality. Thoughts?

Reply:
Yeah, I agree, we can pause on this particular point because it'll get picked up when we step back and look at the build directory as a larger revisit.